### DISCUSSION TITLE
Timeout when opening a new channel after channel exception

### DISCUSSION THREAD

**Discussion by pabermod** [rabbitmq/rabbitmq-dotnet-client#1245] [General]
Hi, I have the following scenario:

- Connection configured with publisher acks.
- Auto recover connection is enabled
- With an open connection and IModel I try to send a message bigger than the default message max size with the method BasicPublish
- This method does not fail. Instead the event ModelShutdown of the IModel is fired.

I have a handler for that event that opens a new channel for the same connection. When openning the new channel I get the following error:

> System.TimeoutException: Se excedió el tiempo de espera de la operación.
   en RabbitMQ.Util.BlockingCell\`1.WaitForValue(TimeSpan timeout)
   en RabbitMQ.Client.Impl.SimpleBlockingRpcContinuation.GetReply(TimeSpan timeout)
   en RabbitMQ.Client.Impl.ModelBase.ModelRpc(MethodBase method, ContentHeaderBase header, Byte[] body)
   en RabbitMQ.Client.Framing.Impl.Model._Private_ChannelOpen(String outOfBand)
   en RabbitMQ.Client.Framing.Impl.AutorecoveringConnection.CreateNonRecoveringModel()
   en RabbitMQ.Client.Framing.Impl.AutorecoveringConnection.CreateModel()`

I attach .Net project example for the problem that uses:
- Net461 (but also happens in Net6)
- Latest RabbitMQ.Client package
- Latest rabbitmq docker image: docker run --hostname my-rabbit --name some-rabbit -p 5672:5672 -p 15672:15672 rabbitmq:3-management

The example sends the content of the given file. Will close the channel is size is bigger than 134217728 bytes:

[RabbitMQError.zip](https://github.com/rabbitmq/rabbitmq-dotnet-client/files/9476036/RabbitMQError.zip)

**Comment by michaelklishin** [comment]
See server logs for clues. Both channel and connection errors are reported by the server asynchronously, so the client can try another operation after one that fails and will be notified in a brief moment. That operation, assuming it expects a server response, will time out.

**Comment by pabermod** [comment]
RabbitMQ log
![image](https://user-images.githubusercontent.com/9059614/187892044-015da856-d3a4-477e-b86c-701d2562a0e1.png)

My event handler to ModelShutdown:
![image](https://user-images.githubusercontent.com/9059614/187892196-ebe61af1-ddb0-41c7-8715-1ff367979792.png)

When the event handler is executed, The connection is still open... So I expect I can create a new Model... But I get the timeout exception. If this is not supported it should be documented

**Reply by pabermod** [reply]
The channel is created (i can see it in the management console) but the .net client generates a timeout exception

**Comment by lukebakken** [comment]
@pabermod we *must* know the versions of software you're using -
* Erlang
* RabbitMQ
* RabbitMQ .NET client
* .NET

In addition, what will help us help you the fastest is if you provide code that I can clone, compile and run to observe this issue. By not providing code you are asking me to take extra time _guessing_ how to reproduce what you report when instead I could be using that time to diagnose your issue.

**Reply by pabermod** [reply]
Hi, I have edited the OP with versions and an example VS project

**Reply by lukebakken** [reply]
Thank you, I converted your code to a .NET 6 project here - https://github.com/lukebakken/rabbitmq-dotnet-client-1245

I can reproduce the issue and will investigate when I have time.

**Reply by lukebakken** [reply]
Please see #1246

**Comment by pabermod** [comment]
I will provide an example tomorrow

Obtener Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by lukebakken** [reply]
I appreciate it! No rush at all.

**Comment by lukebakken** [comment]
#1246 1246

**Comment by lukebakken** [comment]
https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1246#issuecomment-1239995367
